### PR TITLE
Schneems/backport evented file boot at check time

### PR DESCRIPTION
**Pull request by schneems**
Backport of https://github.com/rails/rails/pull/25302

Even though https://github.com/rails/rails/pull/25302 had a 5-0 milestone, it was never backported to the branch :(

Having master not be the current pre-release target is error prone.

### CODE REVIEW

File: activesupport/test/evented_file_update_checker_test.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def setup
   end
 
   def new_checker(files = [], dirs = {}, &block)
-    ActiveSupport::EventedFileUpdateChecker.new(files, dirs, &block).tap do
+    ActiveSupport::EventedFileUpdateChecker.new(files, dirs, &block).tap do |c|
```
Comment (iffyuva) [inline activesupport/test/evented_file_update_checker_test.rb]: hjjj

**Comment by schneems** [comment]
cc @matthewd @sgrif

**Comment by rafaelfranca** [comment]
> Having master not be the current pre-release target is error prone.

This is being what we do for every release 😄. master is not the 4-2-stable release target for years already.

**Comment by rafaelfranca** [comment]
It seems backported to me https://github.com/rails/rails/commit/2a71f0213cda482ed0951c751d0d54b092f25be0

**Comment by schneems** [comment]
Weird. I see the change when viewing the file directly in the branch, however the diff on this PR doesn't seem to reflect that, not sure why.

**Comment by matthewd** [comment]
@schneems when you rebased your change onto the branch, you did it onto a very old commit. GitHub shows the changes on the to-be-merged branch since the common branch point; not a direct diff against the current branch head.

**Pull request closed without merging**